hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -376,6 +376,7 @@ public void run() {
                     handleAuthenticationFailure(endpoint, request);
                 }
             } catch (Throwable e) {
+                logProcessingFailure(request, e);
```

Comment:
I pulled out the logging because I most probably need the `handleProcessingFailure()` method on its own in the next PR. Even if not I don't think this will hurt if I don't use it in the future.